src/storage.ts

```diff
@@ -163,6 +177,77 @@ export type GetHmacKeysResponse = [HmacKey[]];
 
 export const PROTOCOL_REGEX = /^(\w*):\/\//;
 
+/**
+ * Default behavior: Automatically retry retriable server errors.
+ *
+ * @const {boolean}
+ * @private
+ */
+const AUTO_RETRY_DEFAULT = true;
+
+/**
+ * Default behavior: Only attempt to retry retriable errors 3 times.
+ *
+ * @const {number}
+ * @private
+ */
+const MAX_RETRY_DEFAULT = 3;
+
+/**
+ * Default behavior: Wait twice as long as previous retry before retrying.
+ *
+ * @const {number}
+ * @private
+ */
+const RETRY_DELAY_MULTIPLIER_DEFAULT = 2;
+
+/**
+ * Default behavior: If the operation doesn't succeed after 600 seconds,
+ *  stop retrying.
+ *
+ * @const {number}
+ * @private
+ */
+const TOTAL_TIMEOUT_DEFAULT = 600;
+
+/**
+ * Default behavior: Wait no more than 64 seconds between retries.
+ *
+ * @const {number}
+ * @private
+ */
+const MAX_RETRY_DELAY_DEFAULT = 64;
+
+/**
+ * Returns true if the API request should be retried, given the error that was
+ * given the first time the request was attempted.
+ * @const
+ * @private
+ * @param {error} err - The API error to check if it is appropriate to retry.
+ * @return {boolean} True if the API request should be retried, false otherwise.
+ */
+const RETRYABLE_ERR_FN_DEFAULT = function (err?: ApiError) {
+  if (err) {
+    if ([408, 429, 500, 502, 503, 504].indexOf(err.code!) !== -1) {
+      return true;
+    }
+
+    if (err.errors) {
+      for (const e of err.errors) {
+        const reason = e.reason?.toLowerCase();
+        if (
+          (reason && reason.includes('eai_again')) ||
+          reason === 'connection reset by peer' ||
```

Comment:
Is it same to assume the text here won't change?

Reply:
There's nothing stopping the GCS team from changing it. Similar logic has been in the released version of nodejs-common (https://github.com/googleapis/nodejs-common/blob/513374ca1bce94ba720106ab78c776a0e61c0358/src/util.ts#L549) and I'm not sure of another way to check for this error. This logic was designed to be similar to common.